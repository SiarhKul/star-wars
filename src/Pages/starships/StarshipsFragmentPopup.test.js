import { shallow } from "enzyme";
import React from "react";
import { StarshipsFragmentPopup } from "./StarshipsFragmentPopup";

jest.mock("history", () => {
	return {
		createBrowserHistory: jest.fn(() => {
			return {
				location: {
					state: {
						name: "Skol",
						model: "4f-B",
						manufacturer: "Red Planet",
						cost_in_credits: "44400000",
						length: "300000",
						max_atmosphering_speed: "456",
						cargo_capacity: "23",
						consumables: "0",
						hyperdrive_rating: "444",
						MGLT: "NF-3",
						starship_class: "Fast",
						people: ["https://swapi.dev/api/people/15/"],
						films: ["https://swapi.dev/api/films/1/"],
					},
				},
			};
		}),
	};
});

describe("Test <StarshipsFragmentPopup/>", () => {
	const starshipsFragmentPopup = () => shallow(<StarshipsFragmentPopup />);
	let component;

	beforeEach(() => {
		component = starshipsFragmentPopup();
	});

	it("render snapshot  <StarshipsFragmentPopup/>", () => {
		expect(component).toMatchSnapshot();
	});

	it("should  <StarshipsFragmentPopup/>  has required fields", () => {
		const requiredFields = [
			"Model:",
			"Manufacturer:",
			"Cost in credits:",
			"length:",
			"Max atmosphering speed:",
			"Passengers:",
			"Consumables:",
			"Hyperdrive rating:",
			"MGLT:",
			"Starship class:",
		];
		expect(component.containsAllMatchingElements(requiredFields)).toEqual(true);
	});
});
