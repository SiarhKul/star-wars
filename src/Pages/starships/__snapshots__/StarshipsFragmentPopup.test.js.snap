// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <StarshipsFragmentPopup/> render snapshot  <StarshipsFragmentPopup/> 1`] = `
<Fragment>
  <div
    className="name-wrapper"
  >
    <span
      className="popup__name"
    >
      Skol
    </span>
  </div>
  <div
    className="description-wrapper"
  >
    <div
      className="description-person"
    >
      <div
        className="description-appearance"
      >
        <p
          className="description__heading"
        >
          Appearance
        </p>
        <ul
          className="description-appearance__item"
        >
          <li>
            Model:
            4f-B
          </li>
          <li>
            Manufacturer:
            Red Planet
          </li>
          <li>
            Cost in credits:
            44400000
          </li>
          <li>
            length:
            300000
          </li>
          <li>
            Max atmosphering speed:
            456
          </li>
          <li>
            Passengers:
            23
          </li>
          <li>
            Consumables:
            0
          </li>
          <li>
            Hyperdrive rating:
            444
          </li>
          <li>
            MGLT:
            NF-3
          </li>
          <li>
            Starship class:
            Fast
          </li>
        </ul>
      </div>
    </div>
    <div
      className="description-resources"
    >
      <div
        className="resources"
      >
        <div
          className="resources-heading"
        >
          Pilots:
        </div>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="name"
            path="people"
          />
        </ul>
      </div>
      <div
        className="resources"
      >
        <div
          className="resources-heading"
        >
          Films:
        </div>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="title"
            path="films"
            urls={
              Array [
                "https://swapi.dev/api/films/1/",
              ]
            }
          />
        </ul>
      </div>
    </div>
  </div>
</Fragment>
`;
